paasta_tools/utils.py

```diff
@@ -1551,6 +1552,80 @@ def list_clusters(service=None, soa_dir=DEFAULT_SOA_DIR, instance_type=None):
     return sorted(clusters)
 
 
+def get_instance_configs(instance_type, cluster=None, soa_dir=DEFAULT_SOA_DIR):
+    """Returns instance configs associated with an instance type and cluster
+
+    :param instance_type: The instance type, eg chronos or marathong
+    :param cluster: The cluster to look in.  If not specified will lookup the current cluster
+    :returns: dict in format {service_name: {instance_name: {instance_conf}, ...}, ...}
+    """
+    if not cluster:
+        cluster = load_system_paasta_config().get_cluster()
+    rootdir = os.path.abspath(soa_dir)
+    service_instance_configs = {}
+    for srv_dir in os.listdir(rootdir):
+        service_instance_configs[srv_dir] = service_configuration_lib.read_extra_service_information(
+            srv_dir,
+            "%s-%s" % (instance_type, cluster),
+            soa_dir=soa_dir,
+        )
+    return service_instance_configs
+
+
+def get_all_instance_configs(clusters, soa_dir=DEFAULT_SOA_DIR):
+    """Get all instance configs for all instance types for a list of clusters
+
+    :param clusters: The clusters to get configs for
+    :returns: A dict in the format {cluster_name: {service_name: {instance_name: {instance_conf}, ...}, ...}, ...}
+    """
+    confs = defaultdict(dict)
+    for t in INSTANCE_TYPES:
+        for c in clusters:
+            services = get_instance_configs(t, c, soa_dir)
+            for service, instance in services.items():
+                if service not in confs[c].keys():
```

Comment:
That doesn't seem to work, since `defaultdict` takes a type and `defaultdict(dict)` is not callable.  I think I can actually fix this though with `defaultdict(lambda: defaultdict(dict))`.  From here: https://stackoverflow.com/questions/9358983/python-dictionary-and-default-values/23933745#23933745